src/BoundaryConditions/flux_conditions.jl

```diff
@@ -19,6 +19,16 @@ struct CustomFluxCondition <: AbstractBoundaryCondition
 end
 
 """
-    DragLawCondition <: AbstractFluxBoundaryCondition
+    DragLawCondition{C} <: AbstractFluxBoundaryCondition
 """
-struct DragLawCondition <: AbstractBoundaryCondition end
+mutable struct DragLawCondition{C} <: AbstractBoundaryCondition
```

Comment:
someone would know better, but you can also make the struct immutable as long as the containers it has are mutable (like e.g. an array) - then you can update struct.container_contents. Not sure of the pros/cons to each approach though or if it matters!

Reply:
👍 on it... :)

Reply:
> someone would know better, but you can also make the struct immutable as long as the containers it has are mutable (like e.g. an array)

For them to reside on the GPU, they need to be stack-allocated, that's why we used `SVector`s in ClimateMachine.jl's `AtmosModel`. Mutable arrays (specifically, `MArray`s) _can_ reside on the GPU, assuming that the compiler can infer that they don't escape the kernel. But that was typically only done in the ClimateMachine.jl's backend RHS kernels.

All that said, the solution currently implemented looks fine to me.